ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13707,14 +13709,25 @@ ASTNode analyzeCreateTable(
 
   /** Adds entities for create table/create view. */
   private void addDbAndTabToOutputs(String[] qualifiedTabName, TableType type,
-      boolean isTemporary, Map<String, String> tblProps) throws SemanticException {
+      boolean isTemporary, Map<String, String> tblProps, StorageFormat storageFormat) throws SemanticException {
     Database database  = getDatabase(qualifiedTabName[0]);
     outputs.add(new WriteEntity(database, WriteEntity.WriteType.DDL_SHARED));
 
     Table t = new Table(qualifiedTabName[0], qualifiedTabName[1]);
     t.setParameters(tblProps);
     t.setTableType(type);
     t.setTemporary(isTemporary);
+    HiveStorageHandler storageHandler = null;
+    try {
+      storageHandler = (HiveStorageHandler) ReflectionUtils.newInstance(
+              conf.getClassByName(storageFormat.getStorageHandler()), SessionState.get().getConf());
+    }catch(ClassNotFoundException ex){
+      System.out.println("Class not found. Storage handler will be set to null: "+ex);
+    }
+    t.setStorageHandler(storageHandler);
+    for(Map.Entry<String,String> serdeMap : storageFormat.getSerdeProps().entrySet()){
```

Comment:
is there a reason we are setting serde params on the table params? can you give me an example of why we want to do this?

Reply:
Currently, in the compilation phase we are not sending serde information for ranger auth. This storageFormat's serde has table mappings(eg: hbase table name, column family) of external that needs to be set to table serde.